tt_eager/tt_dnn/op_library/moreh_nll_loss_unreduced_backward/moreh_nll_loss_unreduced_backward_op.cpp

```diff
@@ -0,0 +1,172 @@
+    const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+    const std::vector<std::optional<Tensor>>& output_tensors) const {
+    TT_FATAL(input_tensors.size() == 2, "Must have 2 input tensors");
+    TT_FATAL(optional_input_tensors.size() == 1, "Must have 1 optional input tensors");
+
+    auto& target_tensor = input_tensors.at(0);
+    auto& output_grad_tensor = input_tensors.at(1);
+
+    auto& weight_tensor = optional_input_tensors.at(0);
+
+    auto& input_grad_tensor = output_tensors.at(0);
+
+    TT_FATAL(target_tensor.storage_type() == StorageType::DEVICE, "Operands to nll_loss_unreduced need to be on device!");
+    TT_FATAL(target_tensor.buffer() != nullptr, "Operands to nll_loss_unreduced need to be allocated in buffers on device!");
+    TT_FATAL((target_tensor.get_layout() == Layout::TILE), "target_tensor to nll_loss_unreduced must be tilized");
+    TT_FATAL(target_tensor.get_dtype() == DataType::INT32);
+
+    TT_FATAL(output_grad_tensor.storage_type() == StorageType::DEVICE, "Operands to nll_loss_unreduced need to be on device!");
+    TT_FATAL(
+        output_grad_tensor.buffer() != nullptr, "Operands to nll_loss_unreduced need to be allocated in buffers on device!");
+    TT_FATAL((output_grad_tensor.get_layout() == Layout::TILE), "target_tensor to nll_loss_unreduced must be tilized");
+    TT_FATAL(output_grad_tensor.get_dtype() == DataType::BFLOAT16);
+
+    if (input_grad_tensor.has_value()) {
+        TT_FATAL(
+            input_grad_tensor.value().storage_type() == StorageType::DEVICE,
+            "Operands to nll_loss need to be on device!");
+        TT_FATAL(
+            input_grad_tensor.value().buffer() != nullptr,
+            "Operands to nll_loss need to be allocated in buffers on device!");
+        TT_FATAL(
+            (input_grad_tensor.value().get_layout() == Layout::TILE), "target_tensor to nll_loss_unreduced must be tilized");
+        TT_FATAL(input_grad_tensor.value().get_dtype() == DataType::BFLOAT16);
+    }
+
+    if (weight_tensor.has_value()) {
+        TT_FATAL(
+            weight_tensor.value().storage_type() == StorageType::DEVICE,
+            "weight_tensor to nll_loss need to be on device!");
+        TT_FATAL(
+            weight_tensor.value().buffer() != nullptr,
+            "weight_tensor to nll_loss need to be allocated in buffers on device!");
+        TT_FATAL((weight_tensor.value().get_layout() == Layout::TILE), "weight_tensor to nll_loss_unreduced must be in tilized");
+        TT_FATAL(weight_tensor.value().get_dtype() == DataType::BFLOAT16);
+    }
+}
+
+std::vector<Shape> MorehNllLossUnreducedBackward::compute_output_shapes(const std::vector<Tensor>& input_tensors) const {
+    // To calculate the output shape, we need the channel_size. However, the required tensors, target and output_grad,
+    // do not contain the channel_size information.
+    TT_FATAL(false, "moreh_nll_loss_unreduced_backward not support create output tensors.");
```

Comment:
> not support create output tensors

A bit unexpected to see this inside `compute_output_shapes`

Reply:
Are your team thinking about the scenario that the end-user directly call the backward ops?